Resources/views/Form/iconpicker.html.twig

```diff
@@ -0,0 +1,34 @@
+{#
+
+This file is part of the Sonata package.
+
+(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+
+#}
+{% block sonata_type_icon_selector_widget %}
+    {{ block('choice_widget') }}
+    {% spaceless %}
+        <script type="text/javascript">
+            jQuery(function ($) {
+                var select2FormatColorSelect = function format(state) {
+                    if (!state.id || state.disabled) {
+                        return state.text;
+                    }
+
+                    return ' <i class="fa '+ state.text +'"></i> ' + state.text;
```

Comment:
> Not a big fan of twig generating js

Yes please, please, please don't do this. :pray: 
Put all your javascript in one external javascript file.
Make use of event delegation and pass data via `data-` attributes.

```twig
{% block sonata_type_icon_selector_widget %}
  <div class="sonata-dashboard-icon-selector" data-field-id="{{ id }}">
    {{ block('choice_widget') }}
  </div>
{% endblock %}
```